src/Security/Certificate.cs

```diff
@@ -551,11 +551,19 @@ public SecKey (IntPtr handle, bool owns)
 		[DllImport (Constants.SecurityLibrary, EntryPoint="SecKeyGetTypeID")]
 		public extern static nint GetTypeID ();
 		
+		[Deprecated (PlatformName.MacOSX, 12, 0, message: "Use 'SecKeyCreateRandomKey' instead.")]
+		[Deprecated (PlatformName.iOS, 15, 0, message: "Use 'SecKeyCreateRandomKey' instead.")]
+		[Deprecated (PlatformName.TvOS, 15, 0, message: "Use 'SecKeyCreateRandomKey' instead.")] 
+		[Deprecated (PlatformName.WatchOS, 8, 0, message:  "Use 'SecKeyCreateRandomKey' instead.")]
```

Comment:
You need to use the new (dotnet) attributes in manual bindings. Otherwise you're pushing extra work to fix https://github.com/xamarin/xamarin-macios/issues/11055

```csharp
#if NET
		[UnsupportedOSPlatform ("macos12.0")]
		[UnsupportedOSPlatform ("ios15.0")]
		[UnsupportedOSPlatform ("tvos15.0")]
		[UnsupportedOSPlatform ("maccatalyst15.0")]
#else
		[Deprecated (PlatformName.MacOSX, 12, 0, message: "Use 'SecKeyCreateRandomKey' instead.")]
 		[Deprecated (PlatformName.iOS, 15, 0, message: "Use 'SecKeyCreateRandomKey' instead.")]
 		[Deprecated (PlatformName.TvOS, 15, 0, message: "Use 'SecKeyCreateRandomKey' instead.")] 
 		[Deprecated (PlatformName.WatchOS, 8, 0, message:  "Use 'SecKeyCreateRandomKey' instead.")]
#endif
```

No need for `[UnsupportedOSPlatform ("watchos")]` since it's done at the assembly-level, see `src/MinimumVersions.cs.in`

However you need `[UnsupportedOSPlatform ("maccatalyst15.0")]` or `[UnsupportedOSPlatform ("maccatalyst")]` (and `#if !MACCATALYST`) if it has never been available.

Optionally you can add an `[Obsolete]` attribute to suggest an alternative. This is what the generator does, see `src/ObjCRuntime/PlatformAvailability2.cs`.

The tricky part is that it's general to all platforms, so you might need to use `#if ... #endif` around each `[Obsolete]` if the message is not identical (or if it's not deprecated on every platforms).